LandSandBoat/server
# [sql/lua] fix A_THIEF_IN_NORG entry req and completion

**_I affirm:_**

## What does this pull request do?

First, simple error in the entry req. can easily check the function logic works by running the exec manually when on the quest:
`xi.quest.getVar(player, xi.questLog.OUTLANDS, xi.quest.id.outlands.A_THIEF_IN_NORG, 'Prog')`

Next, the `Gaki` mob is a smn, with `Ayakashi` as an astral flow pet and `Onibi` as a spirit pact. The spawn positions made battlefield 1 impossible to complete due to the spirit being far far away. updated spawn point of spirit and carbuncle to be near the smn

Updated the zone table mob entry to be the table of Gaki mobs to clean up the look of the battlefield lua. 

This made things all wacky. I realized that the order of the table was bad, due to no sort on the query:

![image](https://github.com/LandSandBoat/server/assets/131182600/5d4d905b-c683-42df-8db4-fccb138a145d)

updated that so that index 1 is reliably the lowest id

## Steps to test these changes

`!togglegm` to get right into the battlefield, or do the legwork to test the entry req fix by changing job to sam, completiong prereq, bouncing around to get the 6 required actions (talk to npcs, zone in, etc)

confirm 3 mobs and spirit spawn, use `!exec target:useMobAbility(xi.jsa.ASTRAL_FLOW)` to see the carbuncle join the fray
